compiler/rustc_trait_selection/src/solve/normalize.rs

```diff
@@ -41,6 +42,19 @@ pub(crate) fn deeply_normalize_with_skipped_universes<'tcx, T: TypeFoldable<TyCt
     value.try_fold_with(&mut folder)
 }
 
+// Deeply normalize a value and return it
+pub(crate) fn deeply_normalize_for_diagnostics<'tcx, T: TypeFoldable<TyCtxt<'tcx>>>(
+    infcx: &InferCtxt<'tcx>,
+    param_env: ty::ParamEnv<'tcx>,
+    t: T,
+) -> T {
+    infcx
```

Comment:
can you add an assert that this is only used with the new solver or alternatively, make it a noop if it isn't?

Reply:
I think we already assert if `deep_normalize` is called outside of the new solver, but I'll copy that assertion here to really make it clear 😃